metapool/literals.py

```diff
@@ -0,0 +1,112 @@
+# I am aware that "having a global constants class" is largely regarded as
+# an antipattern in Python because it increases dependence between the
+# modules using it and keeps pieces tightly coupled that should be loosely
+# coupled.  However, the current codebase has some literal strings repeated
+# *throughout* the code, and I need them all in one place while we get them
+# replaced with constants so that we can (a) prevent re-definition of the same
+# constant string in multiple places and (b) get a clear view of which of
+# them are really cross-domain and which really do belong to a specific
+# domain (where they can then be moved in the future).  Constants that I
+# already *know* to be domain-specific are not included here.
+
+# Note:
+# A number of these string literals are also used in amplicon-related code
+# (e.g., prep.py, scripts/seqpro.py, seqpro_mf.py, etc.).  However, I don't
+# know if they are the same because they come from the same source, or if it is
+# just coincidental.  For now, I have NOT changed the amplicon-related
+# instances to use these constants.
+
+import re
+
+# These are the "standard"-format snake-case column names;
+# if writing something NEW, use these
+SAMPLE_NAME_KEY = "sample_name"
+QIITA_ID_KEY = "qiita_id"
+SAMPLE_PROJECT_KEY = "sample_project"
+ORIG_NAME_KEY = "orig_name"
+EXPT_DESIGN_DESC_KEY = 'experiment_design_description'
+CONTAINS_REPLICATES_KEY = "contains_replicates"
+PROJECT_SHORT_NAME_KEY = "short_project_name"
+PROJECT_FULL_NAME_KEY = "full_project_name"
+# NB: don't use this as the name for a column of sample identifiers.  It is
+# the key for a list of sample info *objects* (e.g., dictionaries)
+SAMPLES_KEY = "samples"
+
+# Plate map (PM) column names
+PM_SAMPLE_KEY = "Sample"
+PM_PROJECT_PLATE_KEY = "Project Plate"
+PM_PROJECT_NAME_KEY = "Project Name"
+PM_PROJECT_ABBREV_KEY = 'Project Abbreviation'
+PM_COMPRESSED_PLATE_NAME_KEY = "Compressed Plate Name"
+PM_BLANK_KEY = "Blank"
+PLATE_NAME_DELIMITER = "_"
+
+# Currently used to describe sample context (only)
+SAMPLE_TYPE_KEY = "sample_type"
+PRIMARY_STUDY_KEY = "primary_qiita_study"
+SECONDARY_STUDIES_KEY = "secondary_qiita_studies"
+
+
```

Comment:
Hmm, I agree that it is always hard to come up with good nomenclature.  All the contents of this module--both the constants and the functions--are centered around *string names*.  The constants codify some string names that are used in various places throughout the code, and the functions encode knowledge of the naming conventions for other string names used throughout the code (and perform string operations on these names based on the encoded knowledge).  This allows code outside this module to e.g. break apart the name of a project into its component pieces *without* having to have any knowledge of how the pieces are actually represented in the name, because both the knowledge of the expected format and the constants defining expected delimiters are all bundled together in here.  I am happy to change the name of this module to something that would better represent its mission (and very open to suggestions for what that would be ... :D )

Reply:
Hmm, in my experience this type of file is usually called constants.py and doesn't contain functions. However that's not really any more descriptive than literals.py in this case.

Although 'strings' is not a reserved word, IMHO it's too close to the built-in 'string' module. The string module also has helper functions though. Maybe something analogous like mpstrings.py, mp_strings.py, MPStrings.py?

Reply:
sold!

Reply:
ty!